### PR TITLE
add instruction about how to install OF from github

### PR DESCRIPTION
**Pull request by edap**
After some discussion in the forum (https://forum.openframeworks.cc/t/compile-of-locally/26248 , https://forum.openframeworks.cc/t/how-do-i-build-from-source/24569/2, https://forum.openframeworks.cc/t/how-to-build-project-generator-from-of-git-repo/26232 and many others), I've written this.

I've followed this procedure, and now in my OF folder i've two changed files:

```bash
 M apps/projectGenerator
 M scripts/apothecary
```

Because I've run previously `git submodule update`. This is the output of git diff

```
diff --git a/apps/projectGenerator b/apps/projectGenerator
index 2b24ac0..2a1cb6a 160000
--- a/apps/projectGenerator
+++ b/apps/projectGenerator
@@ -1 +1 @@
-Subproject commit 2b24ac02060254855bc3a7560aed64d1c97a4ba9
+Subproject commit 2a1cb6af9c6a957b84d645136c996990a3c02726
diff --git a/scripts/apothecary b/scripts/apothecary
index d05cfc8..8c43fb3 160000
--- a/scripts/apothecary
+++ b/scripts/apothecary
@@ -1 +1 @@
-Subproject commit d05cfc8057131aaa8289b9a003be74c9c2c1f985
+Subproject commit 8c43fb32ebd849ac32e98574bec9e62488accbe3-dirty
```
How should i deal with these modified files? I would address this situation in this readme too.

### CODE REVIEW

File: INSTALL_FROM_GITHUB.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## How to compile openFrameworks from github
+
+This instructions are useful just in case you want to submit a pull request or you want to have always the last openFrameworks version. In any other case, most probably, you don't need to clone the entire openFrameworks repository.
+
+If you are a developer, or if you want to submit a pull request, [read this first](https://github.com/openframeworks/openFrameworks/blob/master/README.md#developers)
+
+Let's start cloning the last `stable` branch of openframeworks.
+
+```bash
+git clone git@github.com:openframeworks/openFrameworks.git
```
Comment (arturoc) [inline INSTALL_FROM_GITHUB.md]: you can download the submodules right away if you use:

```bash
git clone --recursive git@github.com:openframeworks/openFrameworks.git
```

File: INSTALL_FROM_GITHUB.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Now you have all the files that you need, you just need to compile them and to setup all the projects contained in the `examples` folder. The easiest way to do this is to use the projectGenerator to compile all the examples.
+The projectGenerator is useful app to create new project and to add and remove addons. As it is a separate project, with a well defined scope, the openFrameworks developer decided to put it in a separate repository. You have 2 ways to get the project generator in your openFrameworks installation:
+* You can initialize and update all the git submodules (including the projectGenerator) running first `git submodule init` and then `git submodule update`
+* Or, more easily, you can download an already packaged openFrameworks installation from the [download page](http://openframeworks.cc/download/), and copy the `projectGenerator` folder in your openFrameworks installation.
+
+As the first option can lead to [unexpected results](https://forum.openframeworks.cc/t/how-to-build-project-generator-from-of-git-repo/26232), we suggest you the second one.
+
+Now that you have the project generator, you are ready to compile all the examples. Do as follow:
+
```
Comment (arturoc) [inline INSTALL_FROM_GITHUB.md]: there's already instructions for the PG perhaps just link to them from here so this the minimum necessary only to setup from git? also if there's changes in the PG we don't need to update multiple places: http://openframeworks.cc/learning/01_basics/create_a_new_project/

File: INSTALL_FROM_GITHUB.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+* Then select the `update multiple` tab, put in the path to the examples folder in the current OpenFrameworks repo folder.
+* Make sure to select your platform in the platforms field below.
+* Hit `update multiple` and wait. This operation can take a while depending on your hardware.
+
+Once the success message appears you are done.
+
+
+## How to submit you pull requests
+
+The procedure to install openFrameworks locally is almost the same, but you neef first to fork the openFrameworks repository from github, and clone on your local machine your fork of the openFrameworks repository instead `git@github.com:openframeworks/openFrameworks.git`. Please be sure to read [this](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) before starting.
```
Comment (arturoc) [inline INSTALL_FROM_GITHUB.md]: says neef, should say need

File: INSTALL_FROM_GITHUB.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+## How to compile openFrameworks from github
+
+This instructions are useful just in case you want to submit a pull request or you want to have always the last openFrameworks version. In any other case, most probably, you don't need to clone the entire openFrameworks repository.
+
+If you are a developer, or if you want to submit a pull request, [read this first](https://github.com/openframeworks/openFrameworks/blob/master/README.md#developers)
+
+Let's start cloning the last `stable` branch of openframeworks.
```
Comment (edap) [inline INSTALL_FROM_GITHUB.md]: addressed in cfa453903c8a1f5719f50d8f2f3c2b68ace69f61

File: INSTALL_FROM_GITHUB.md (hunk 1 above)
Comment (edap) [inline INSTALL_FROM_GITHUB.md]: addressed in https://github.com/openframeworks/openFrameworks/commit/cfa453903c8a1f5719f50d8f2f3c2b68ace69f61

File: INSTALL_FROM_GITHUB.md (hunk 2 above)
Comment (edap) [inline INSTALL_FROM_GITHUB.md]: I see. But the instructions you have linked explains how to use the project generator to create one project. Not how to compile all the examples. Maybe I should move this part to the page you have linked?

File: INSTALL_FROM_GITHUB.md (hunk 3 above)
Comment (edap) [inline INSTALL_FROM_GITHUB.md]: fixed in cfa453903c8a1f5719f50d8f2f3c2b68ace69f61

File: INSTALL_FROM_GITHUB.md (hunk 2 above)
Comment (arturoc) [inline INSTALL_FROM_GITHUB.md]: oh i see, no i think it's fine like this then.

**Comment by arturoc** [comment]
thanks!

**Pull request merged**